src/Utilities/ProgramUtils.tsx

```diff
@@ -24,6 +24,14 @@ const extractGameNameFromProgram = pc => {
   }
 };
 
+/*
+ * removeComments
+ *
+ * Takes a program string, removes all block comments, then single line comments, then trims leading & trailing whitespace
+ */
+const removeComments = pc => pc.replaceAll(/{-(?:.|\n)*-}/g,'').replaceAll(/--[^\n]*/g,'').trim();
```

Comment:
Factored this out to multi-line and single-line comments can be screened pretty easily. Multi has to come before single, otherwise some comments like `{-- --}` would get mangled with a leftover `{`. Granted, this is never passed to the backend, and it's just used to eliminate information we don't care for when extracting names and such, but it's good to have it also produce a correct program.